src/etcd_client.rs

```diff
@@ -53,21 +45,14 @@ impl EtcdClient {
         key: &str,
         value: &T,
     ) -> anyhow::Result<Option<T>> {
-        let bin_value = bincode::serialize(value).map_err(|e| {
-            anyhow!(
-                "failed to encode {:?} to binary, the error is: {}",
-                value,
-                e,
-            )
-        })?;
+        let bin_value = bincode::serialize(value)?;
         let mut req = etcd_rs::PutRequest::new(key, bin_value);
         req.set_prev_kv(true); // Return previous value
         let mut resp = self
             .etcd_rs_client
             .kv()
             .put(req)
-            .await
-            .map_err(|e| anyhow!("failed to get PutResponse from etcd, the error is: {}", e))?;
+            .await?;
```

Comment:
add `.context(format!("failed to get PutResponse from etcd for PutRequest={:?}", req))?;`